test/parallel/test-buffer.js

```diff
@@ -749,6 +749,11 @@ for (let i = 0; i < 256; i++) {
   assert.equal(hexb2[i], hexb[i]);
 }
 
+//#6770 Test single hex character throws TypeError
+assert.throws(function() {
+  Buffer.from('A', 'hex');
+}, TypeError);
```

Comment:
Can you put a test to verify there's no regression to `'base64'` since it follows the same logic path in this case? Something simple like `assert.equal(Buffer.from('A', 'base64'), 0)`.